subsys/bluetooth/controller/ll_sw/ctrl.c

```diff
@@ -1090,7 +1090,7 @@ static inline u32_t isr_rx_adv(u8_t devmatch_ok, u8_t devmatch_id,
 		radio_le_conn_cmplt->own_addr_type = pdu_adv->rx_addr;
 		memcpy(&radio_le_conn_cmplt->own_addr[0],
 		       &pdu_adv->connect_ind.adv_addr[0], BDADDR_SIZE);
-		if (rl_idx != FILTER_IDX_NONE) {
+		if (irkmatch_ok && rl_idx != FILTER_IDX_NONE) {
```

Comment:
A couple of questions though:

1. Maybe better to fix this higher up, when we set `rl_idx` and set it to `NONE` if `irkmatch_ok == FALSE`?
2. Does this need to be fixed in the split LL?

Reply:
1. I'm not 100% sure that would be without side effects. This way matches the inititiaor side of it

2. As far as I could tell the split one handled this, so no